src/policy/policy.h

```diff
@@ -0,0 +1,67 @@
+#ifndef SRC_POLICY_POLICY_H_
+#define SRC_POLICY_POLICY_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_options.h"
+#include "v8.h"
+#include "policy/policy_deny.h"
+#include "policy/policy_deny_fs.h"
+
+#include <map>
+#include <iostream>
+
+namespace node {
+
+class Environment;
+
+namespace policy {
+
+#define THROW_IF_INSUFFICIENT_PERMISSIONS(env, perm_, resource_, ...)          \
+  if (!node::policy::root_policy.is_granted(perm_, resource_)) {               \
+    return node::policy::Policy::ThrowAccessDenied((env), perm_);              \
+  }
```

Comment:
Similar as above, you might want to look at how https://github.com/nodejs/node/pull/22112 implemented the macro of the same name. This applies both to the level at which modifications happen (you can’t check a global mutable value without concurrent access protection) and the performance aspect.